### ISSUE TITLE
VT API requests cause Logstash process to "freeze"

### ISSUE THREAD

**Issue by xme** [coolacid/logstash-filter-virustotal#1]
I configured my Logstash with an extra filter to automatically search for MD5 hashed on VT. The filter is defined as follow:

filter {
    if ( [event_type] == "fileinfo" and [fileinfo][filename] =~ /(?i).(doc|pdf|exe)/ {
        virustotal {
            apikey => "<removed>"
            field => '[fileinfo][md5]'
            lookup_type => 'hash'
            target => 'virustotal'
       }
    }
}

When multiple requests are sent in parallel or with a short delay, the Logstash process enters a stuck state and don't process any new incoming event! (+ all my forwarders are disconnected)

Somebody reported that the VT API is facing some issues for a few days. Is it related?

/x

**Comment by coolacid** [comment]
@xme I just tried running 20 queries via a generator, so it ran really fast. I wonder if this is related to timeout (or maybe there is a lack of a short enough time out)

**Comment by coolacid** [comment]
Latest version has a default 5 second timeout, this is also configurable - would love some tests ;)

**Comment by xme** [comment]
Tx for the tests... I'll retry on my side asap.

**Comment by coolacid** [comment]
I've released a new version if you wouldn't mind testing.

**Comment by coolacid** [comment]
Closing - If still an issue, please open a new issue.
